trunner/runners/common.py

```diff
@@ -343,6 +370,97 @@ def run(self, test):
             self.serial.close()
 
 
+class ARMV7M7Runner(DeviceRunner, ABC):
+    """This class provides interface to run tests on targets with armv7m7 architecture"""
```

Comment:
I think we should put this class in separate file, hopefully it will replace custom runners for RT106x and RT117x in the futre.